nmostesting/suites/BCP0060101Test.py

```diff
@@ -35,110 +38,454 @@ def __init__(self, apis):
         GenericTest.__init__(self, apis, omit_paths)
         self.node_url = self.apis[NODE_API_KEY]["url"]
         self.connection_url = self.apis[CONN_API_KEY]["url"]
+        self.is04_resources = {"senders": [], "receivers": [], "_requested": [], "sources": [], "flows": []}
+
+    # Utility function from IS0502Test
+    def get_is04_resources(self, resource_type):
+        """Retrieve all Senders or Receivers from a Node API, keeping hold of the returned objects"""
+        assert resource_type in ["senders", "receivers", "sources", "flows"]
+
+        # Prevent this being executed twice in one test run
+        if resource_type in self.is04_resources["_requested"]:
+            return True, ""
+
+        valid, resources = self.do_request("GET", self.node_url + resource_type)
+        if not valid:
+            return False, "Node API did not respond as expected: {}".format(resources)
+
+        try:
+            for resource in resources.json():
+                self.is04_resources[resource_type].append(resource)
+            self.is04_resources["_requested"].append(resource_type)
+        except json.JSONDecodeError:
+            return False, "Non-JSON response returned from Node API"
+
+        return True, ""
 
     def test_01(self, test):
         """JPEG XS Flows have the required attributes"""
 
         reg_api = self.apis[FLOW_REGISTER_KEY]
 
-        url = self.node_url + "flows"
-        valid, response = self.do_request("GET", url)
+        valid, result = self.get_is04_resources("flows")
+        if not valid:
+            return test.FAIL(result)
 
         reg_path = reg_api["spec_path"] + "/flow-attributes"
         reg_schema = load_resolved_schema(reg_path, "flow_video_register.json", path_prefix=False)
 
-        if valid and response.status_code == 200:
-            try:
-                flows = response.json()
+        try:
+            found_video_jxsv = False
+            warn_unrestricted = False
+            warn_message = ""
 
-                found_video_jxsv = False
-                warn_unrestricted = False
-                warn_message = ""
+            for flow in self.is04_resources["flows"]:
+                if flow["media_type"] != "video/jxsv":
+                    continue
+                found_video_jxsv = True
```

Comment:
Not too fond of the flag. How about:
`jxsv_flows = [f in self.is04_resources["flows"] if f["media_type"] == "video/jxsv"]`

Then for the loop:
```
for flow in jxsv_flows:
   ...
```

and then at the end of the test (line 125):

```
if len(jxsv_flows) > 0:
   return test.PASS()
```

Reply:
Yeah, could do do that. Had it that way at one point. Will try it...

Reply:
Done